### ISSUE TITLE
Replace current eXist for Roma with updated version

### ISSUE THREAD

**Issue by TEITechnicalCouncil** [TEIC/TEI#527]
The version of eXist which provides the backend to Roma's queries into p5subset.xml needs to be replaced with a current stable version. Indexes will need to be reconfigured (the indexing definition process has changed), and suitable permissions set up; tei-database-update.sh will also need to be tweaked to work with the new setup. This should be done and tested well in advance of the 2.8 release.

Original comment by: @martindholmes

**Comment by lb42** [comment]
Is this ticket still waiting for action?

**Comment by hcayless** [comment]
Since we're looking at sunsetting Roma anyway, this is a very low priority.

**Comment by Conal-Tuohy** [comment]
Is a replacement for Roma under development?

**Comment by lb42** [comment]
Not to my knowledge.

**Comment by martindholmes** [comment]
This project was undertaken, apparently with some success:

https://github.com/TEIC/Byzantium

but I don't think it was finished.

**Comment by Conal-Tuohy** [comment]
Thanks @martindholmes! It certainly looks good on the face of it, but it's hard to tell how "finished" in might be, since the "default TEI database" at http://bits.nsms.ox.ac.uk:8080/jenkins/job/TEIP5/lastSuccessfulBuild/artifact/release/xml/tei/odd/p5subset.js is 404. :-( 

I logged an [issue](https://github.com/TEIC/Byzantium/issues/1).

It looks like the Byzantine approach is to rely on this static artifact in place of the Roman approach of issuing an XQuery to retrieve the current schema. That seems like not such a bad idea, in theory (so long as the js file is where it's supposed to be).

**Comment by jamescummings** [comment]
Bits should be replaced with jenkins.tei-c.org.

There is a Roma replacement planned.  @raffazizzi was looking at it. I prodded him recently about it and do so again through this comment.

**Comment by Conal-Tuohy** [comment]
Thanks @jamescummings!

I tried http://jenkins.tei-c.org:8080/job/TEIP5/lastSuccessfulBuild/artifact/P5/release/xml/tei/odd/p5subset.js (which is 200 OK), but Byzantium still failed to load it. Lending weight to the "not finished" assessment.

I've starred  https://github.com/raffazizzi/romajs -- thanks for the pointer!

I wonder, given the amount of XML expertise in the community, why no-one seems to have written a UI using XForms? (ducks)

**Comment by jamescummings** [comment]
@conal-tuohy Byzantium was only a proof of concept written mostly by an intern with Sebastian and me explaining the TEI side of things.

**Comment by lb42** [comment]
I know of at least one other project (at the University of Caen) which was planning to construct a new Roma on the basis of Byzantium, in their case using xmlMind as the front end. I will ping them to see how far they got and update if there is news.

**Comment by lb42** [comment]
As suspected no progress on the caen project.

**Comment by Conal-Tuohy** [comment]
Anyone interested in Roma, Byzantium, etc, may be interested to follow my project [Muscovy](https://github.com/Conal-Tuohy/Muscovy), an ODD editor that's implemented in XForms. The goal is for it to be purely client-side, with just OxGarage (plus, if it turns out to be necessary, the TEI-C XQuery server) as the back end. 

So far I've spent a while adding missing features to XSLTForms (the XForms implementation), but I've now produced a form that will allow a user to upload a TEI file to OxGarage and receive it in response, validated.
